library/control/src/modules/WorkflowManager.rb

```diff
@@ -415,7 +415,7 @@ def control_file(source)
       package = case source
       when ::Integer
         product = find_product(source)
-        return nil unless product&.product_package
+        return nil if !product.methods.include?(:product_package) || !product&.product_package
```

Comment:
Hm, I have tested it and it has worked. Strange...

Reply:
yes, it is strange, I test it and it does not work for me:

try it yourself in IRB:
```
jreidinger@localhost:~/prace/yast/devtools/package> irb
irb(main):001:0> require "yast"
=> true
irb(main):002:0> require "y2packager/resolvable"
=> true
irb(main):004:0> res = Y2Packager::Resolvable.new("product_package" => "SLES")
irb(main):007:0> res.methods.include?(:product_package)
=> false
irb(main):008:0> res.respond_to?(:product_package)
=> true
```

And why respond_to? works? because rubocop require to have beside `method_missing` also  `respond_to_missing?` which implements dynamic methods for `respond_to?`. see https://github.com/yast/yast-yast2/blob/ab3fee8654cb2b7cca315f9d56b260034fa775dd/library/packages/src/lib/y2packager/resolvable.rb#L130

Reply:
so simply use 
```suggestion
return nil if !product.respond_to?(:product_package) || !product.product_package
```
`&.` is not needed as nil does not respond to product_package method.

Reply:
Thanks Josef, Yes, now it is clear:
irb(main):004:0> res = Y2Packager::Resolvable.new("product_package" => "SLES")
irb(main):007:0> res.methods.include?(:product_package)
In my test it has also returned false where product_package has not been defined. So methods is really the wrong call for it.